R/readNWISunit.r

```diff
@@ -472,4 +480,77 @@ readNWISstat <- function(siteNumbers, parameterCd, startDate = "", endDate = "",
   attr(data, "siteInfo") <- siteInfo
   
   return (data)
-}
\ No newline at end of file
+}
+
+#' Water use data retrieval from USGS (NWIS)
+#' 
+#' Retrieves water use data from USGS Water Use Data for the Nation.  See \url{http://waterdata.usgs.gov/nwis/wu} for 
+#' more information.  All available use categories for the supplied arguments are retrieved. 
+#' 
+#' @param years integer Years for data retrieval — must be years ending in 0 or 5. Default is all available years.
+#' @param  stateAB character Two-letter USPS state abbreviation.  Only one is accepted per query.  Codes can be located
+#' with the stateCdLookup function for US states. Data for the Virgin Islands (VI), Puerto Rico (PR) and the District
+#' of Columbia is also available.  Default is \code{NULL}, which will return national data.
+#' @param county character Name(s) of counties.  Default is \code{NULL}, which will return state or national data 
+#' depending on the stateAB argument.  \code{ALL} may also be supplied, which will return data for every county in a 
+#' state. County names be supplied ending in "County" or not — e.g., "Belmont" and "Belmont County" are both acceptable.  
+#' @param convertType logical defaults to \code{TRUE}. If \code{TRUE}, the function will convert the data to
+#' numerics based on a standard algorithm. Years, months, and days (if appliccable) are also returned as numerics
+#' in separate columns.  If convertType is false, everything is returned as a character.
+#' @return A data frame with at least the year of record, and all available statistics for the given geographic parameters.
+#' County and state fields will be included as appropriate.
+#' 
+#' @export
+#' @importFrom stringr str_sub
+#' @examples 
+#' \dontrun{
+#' #All data for a county
+#' allegheny <- readNWISuse(years="ALL",stateAB = "PA",county = "Allegheny")
+#' 
+#' #Data for an entire state for certain years
+#' ohio <- readNWISuse(years=c(2000,2005,2010),stateAB = "OH")
+#' 
+#' #Data for an entire state, county by county
+#' pr <- readNWISuse(years=c(2000,2005,2010),stateAB = "PR",county="ALL")
+#' 
+#' #All national-scale data
+#' national <- readNWISuse()
+#' 
+#' #Washington, DC data
+#' dc <- readNWISuse(stateAB = "DC")
+#' 
+#' }
+ 
+readNWISuse <- function(years="ALL",stateAB=NULL,county=NULL,convertType=TRUE){
```

Comment:
I'm unsure about using stateCd as a variable name here, since I need to call my stateCdLookup function that uses the stateCd data frame. I've gotten errors with it trying to look up things in the variable, but I can't seem to reproduce it consistently.